### PR TITLE
tests: Use libvmi instead of NewRandomVMIWithEphemeralDisk

### PR DESCRIPTION
**Pull request by bell-lev**
### What this PR does

Use libvmi instead of NewRandomVMIWithEphemeralDisk

Before this PR:

After this PR:

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note

```

### CODE REVIEW

File: tests/dryrun_test.go [hunk 1]
```diff
@@ -67,9 +64,9 @@ var _ = Describe("[sig-compute]Dry-Run requests", decorators.SigCompute, func()
 		var vmi *v1.VirtualMachineInstance
 		resource := "virtualmachineinstances"
 
-		BeforeEach(func() {
-			vmi = tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskAlpine))
-		})
+		vmi = libvmi.NewAlpine()
+		vmi.ObjectMeta.Namespace = testsuite.GetTestNamespace(vmi)
```
Comment (xpivarc) [inline tests/dryrun_test.go:68]: `vmi.ObjectMeta.Namespace = testsuite.GetTestNamespace(vmi)` should not be required.

File: tests/dryrun_test.go [hunk 2]
```diff
@@ -67,9 +64,9 @@ var _ = Describe("[sig-compute]Dry-Run requests", decorators.SigCompute, func()
...
+		vmi.Spec.Domain.Resources.Requests = k8sv1.ResourceList{}
```
Comment (xpivarc) [inline tests/dryrun_test.go:69]: Why is this used?

**Comment by phoracek** [comment]
/ok-to-test

I can vouch for Bell

**Comment by xpivarc** [comment]
> /ok-to-test
> 
> I can vouch for Bell

Please, see a change:

PRs from untrusted users cannot be marked as trusted with /ok-to-test in this repo meaning untrusted PR authors can never trigger tests themselves. Collaborators can still trigger tests on the PR using /test all.

**Comment by phoracek** [comment]
@xpivarc oh gotcha, makes sense.

@bell-lev feel free to ping me offline once you will be ready to run the tests.

**Pull request closed without merging**